### PR TITLE
corrections to make the belongs_to optional in rails 5.1

### PR DESCRIPTION
**Pull request by oniram88**
in the new rails 5.1 version the belongs_to make the association required, to make originator work with applications (like alchemy-cms) that don't require the association of the creator and updater, the change add only the optional: true to the association

it's configurable like the other params, and setup tu be optional as default.

### CODE REVIEW

File: lib/stampable.rb [hunk 1]
```diff
@@ -46,6 +46,11 @@ def self.included(base) #:nodoc:
           # Defaults to :deleter_id when compatibility mode is off
           # Defaults to :deleted_by when compatibility mode is on
           class_attribute  :deleter_attribute
+
+          # Define if the belongs_to association should be optional or not
+          # Defaults to true to mantein the old behavior
```
Comment (tvdeyen) [inline lib/stampable.rb]: Actually the old behaviour of Rails prior to 5.1 is `optional: true`

We should reflect that. Also there is a little typo in the comment ("maintein" vs "maintain")

**Review by tvdeyen — CHANGES_REQUESTED** [review]
Thanks.

We should use `optional: true` as default as this is the default of Rails prior to 5.1

File: originator.gemspec [hunk 2]
```diff
@@ -70,8 +70,8 @@ Gem::Specification.new do |s|
   s.require_paths = ["lib"]
   s.rubygems_version = "1.8.11"
 
-  s.add_runtime_dependency 'actionpack', '>= 4.0', '< 5.1'
-  s.add_runtime_dependency 'activerecord', '>= 4.0', '< 5.1'
+  s.add_runtime_dependency 'actionpack', '>= 4.0'
+  s.add_runtime_dependency 'activerecord', '>= 4.0'
```
Comment (tvdeyen) [inline originator.gemspec]: Could you please make a separate commit out of this change? Thanks

File: lib/stampable.rb (hunk 1 above)
Comment (oniram88) [reply lib/stampable.rb]: and if we let it change to reflect the configuration of the Rails version?
Rails >=5.1 default => false
Rails <5.1 default => true

**Comment by tvdeyen** [comment]
By extra commit I did not meant a separate PR. Instead of adding extra commits for changes requested in a PR review it is good practice to amend the change to the commit where the change is needed and then force push your branch, so the commit history keeps clean.

Anyway, I will cherrypick your changes as I don't want to ask you again to make all this now. Your help is already very much appreciated.

Thanks for that 🍰

**Review by tvdeyen — APPROVED** [review]
